_plugins/series_index.rb

```diff
@@ -88,33 +88,26 @@ def generate(site)
   # Adds some extra filters used during the series creation process.
   module Filters
 
-    # # Outputs a list of seriess as comma-separated <a> links. This is used
-    # # to output the series list for each post on a series page.
-    # #
-    # #  +series+ is the list of series to format.
-    # #
-    # # Returns string
-    # #
+    # Outputs a list of seriess as comma-separated <a> links. This is used
+    # to output the series list for each post on a series page.
+    #
+    #  +series+ is the list of series to format.
+    #
+    # Returns string
+    #
     def series_link(series)
-          require 'pry'
-          binding.pry
-
-      dir = @context.registers[:site].config['series_dir']
-      "<a class='series' href='/#{dir}/#{series.gsub(/_|\P{Word}/, '-').gsub(/-{2,}/, '-').downcase}/'>#{item}</a>"
+      dir = "series"
+      series_id = series.gsub(/_|\P{Word}/, '-').gsub(/-{2,}/, '-').downcase
+      "<a class='series' href='/#{dir}/#{series_id}/'>#{series}</a>"
     end
 
-    # # Outputs the post.date as formatted html, with hooks for CSS styling.
-    # #
-    # #  +date+ is the date object to format as HTML.
-    # #
-    # # Returns string
-    # def date_to_html_string(date)
-    #   result = '<span class="month">' + date.strftime('%b').upcase + '</span> '
-    #   result += date.strftime('<span class="day">%d</span> ')
-    #   result += date.strftime('<span class="year">%Y</span> ')
-    #   result
-    # end
-
+    # Gets a series name, and pulls out a markdown header file
+    # returning either the rendered markdown, or nothing
+    def series_bio(series)
+      series_id = series.gsub(/_|\P{Word}/, '-').gsub(/-{2,}/, '-').downcase
```

Comment:
Quite some repetition going on here with the regex and in `series_link`. I think that extracting it would also be beneficial in documenting what this regexp black magic does exactly.